spec/models/badge_spec.rb

```diff
@@ -271,6 +271,81 @@
     end
   end
 
+  describe "FirstFlag badge" do
+    let(:flagging_user) { Fabricate(:user) }
+    let(:badge_enabled_category) { Fabricate(:category, allow_badges: true) }
+    let(:flagged_post) do
+      Fabricate(:post, topic: Fabricate(:topic, category: badge_enabled_category))
+    end
+    let(:first_flag_badge) { Badge.find(Badge::FirstFlag) }
```

Comment:
```suggestion
    fab!(:flagging_user) { Fabricate(:user) }
    fab!(:badge_enabled_category) { Fabricate(:category, allow_badges: true) }
    fab!(:flagged_post) do
      Fabricate(:post, topic: Fabricate(:topic, category: badge_enabled_category))
    end
    fab!(:first_flag_badge) { Badge.find(Badge::FirstFlag) }
```

I think it would be better if they are fab! for a bit better performance. Not 100% sure if that will work for `:first_flag_badge`